web/js/location.js

```diff
@@ -298,10 +311,44 @@ const getParameters = function(config, parameters) {
       type: 'object',
       initialState: eventsReducerState,
       options: {
-        parse: eventParse,
+        parse: parseEvent,
         serialize: serializeEvent,
       },
     },
+    efs: {
+      stateKey: 'events.showAll',
+      initialState: true,
+      type: 'bool',
+      options: {
+        serializeNeedsGlobalState: true,
+        serialize: (showAll, state) => {
+          const eventsActive = get(state, 'events.active');
+          return eventsActive ? showAll : undefined;
+        },
+        setAsEmptyItem: true,
+      },
+    },
+    efd: {
+      stateKey: 'events.selectedDates',
+      type: 'object',
+      initialState: eventsReducerState.selectedDates,
+      options: {
+        parse: parseEventFilterDates,
+        serialize: serializeEventFilterDates,
+        serializeNeedsGlobalState: true,
+        setAsEmptyItem: true,
+      },
+    },
+    efc: {
```

Comment:
I see all the categories are included in the permalink even though no categories were filtered: http://localhost:3000/?v=-5459977.241413308,-3433509.354667873,5459977.241413308,3433509.354667873&p=arctic&e=EONET_5423,2021-06-22&efs=true&efd=2021-02-23,2021-06-23&efc=dustHaze,manmade,seaLakeIce,severeStorms,snow,volcanoes,waterColor,wildfires&l=IMERG_Precipitation_Rate,VIIRS_SNPP_DayNightBand_ENCC(hidden),VIIRS_SNPP_DayNightBand_At_Sensor_Radiance(hidden),Reference_Labels_15m,Reference_Features_15m,Coastlines_15m(hidden),VIIRS_NOAA20_CorrectedReflectance_TrueColor(hidden),VIIRS_SNPP_CorrectedReflectance_TrueColor,MODIS_Aqua_CorrectedReflectance_TrueColor(hidden),MODIS_Terra_CorrectedReflectance_TrueColor&lg=true&t=2021-06-22-T00%3A00%3A00Z.

Is there anyway to use omit the `efc` parameter if there is no filtering selected?

Reply:
I suppose we could... the way it is now the categories in the URL represent which categories are turned on which, by default, is all of them.  None of these params will show if you don't have the events tab active.  What's the reasoning beyond not showing it when it's in the default state, just to reduce overall URL character length?

Reply:
It seems to me like since All is the default state that no querystring param would be needed kind of like our geographic projection being the default state and omitted. It would be a bonus fix though since as you mentioned it's only displayed when Event tab is active.

Reply:
Bonus achieved!  Only had to tweak a few lines 👍🏻